x-pack/plugins/observability_solution/infra/public/hooks/use_metrics_explorer_views.ts

```diff
@@ -84,7 +91,10 @@ export const useMetricsExplorerViews = (): UseMetricsExplorerViewsResult => {
 
   const { data: currentView, isFetching: isFetchingCurrentView } = useQuery({
     queryKey: queryKeys.getById(currentViewId),
-    queryFn: ({ queryKey: [, id] }) => metricsExplorerViews.client.getMetricsExplorerView(id),
+    queryFn: ({ queryKey: [, id] }) => {
+      return ensureLoadedClient(metricsExplorerViewsClientState).getMetricsExplorerView(id);
+    },
```

Comment:
> [!TIP]
> Adding an async cycle to resolve the client once with `useAsync`, ensure is loaded and have another side effect as a consequence seems unnecessary. Since the client is a singleton instance, you can directly get the client on call site when it's really needed, the client will always be the same and you can rely on an async getter anyway on the query/mutation handlers.
```ts
queryFn: async () => {
  const client = await metricsExplorerViews.getClient();
  return client.findMetricsExplorerViews();
}
```

This applies also to the others queries/mutations in this hook, also applies to the useInventory hook.